Review task1

hgutierrezg opened the pull request.

heidygutierrez committed:
Creating spring mvc skeleton project

Skeleton project from a spring mvc artifact.

Adding git ignore file

heidygutierrez committed:
Adding maven tomcat plugin

heidygutierrez committed:
Angular refactor to add timesheet functionality.
MVC, controllers and service refactor

heidygutierrez committed:
Rename methods
Rename to jsp view
Readme update

heidygutierrez committed:
Readme update

heidygutierrez committed:
Adding form controller

heidygutierrez committed:
Adding list controller

heidygutierrez committed:
Adding list controller

heidygutierrez committed:
Adding .project file to gitignore

Adding .project file to gitignore

heidygutierrez committed:
Emit on create event from create component
List component listens to the event

heidygutierrez committed:
Removed console.log

heidygutierrez committed:
Change to use wrapper classes. Long and Boolean instead of primitives

Mbd06b commented on pom.xml:
```diff
@@ -0,0 +1,105 @@
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-context</artifactId>
+			<version>${spring.version}</version>
+		</dependency>
```
spring-context is duplicated here,  check your dependency hierarchy. spring-webmvc contains spring-context

Mbd06b commented on src/main/java/com/hgutierrezg/training/controller/IndexController.java:
```diff
@@ -0,0 +1,19 @@
+     */
+      @RequestMapping(method = RequestMethod.GET)
+        public String getIndexPage() {
+            return "timesheet";
```
you might want to rename the reference to the page "timesheet" to "timesheetPage", to avoid confusion with a rest endpoint that might "Represent" the timesheet object.

hgutierrezg replied:
This page is removed now, was not needed since index.html is the one being served

Mbd06b commented on src/main/java/com/hgutierrezg/training/model/Timesheet.java:
```diff
@@ -0,0 +1,51 @@
+@NoArgsConstructor
+@AllArgsConstructor
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Timesheet {
```
Refactor this to TimesheetView, later when you need a model for Hibernate, cerate a TimesheetImpl,  both these classes can inherit from an Timesheet interface.

hgutierrezg replied:
This was changed by creating Dto and Model object and mapping them with Mapstruct

Mbd06b commented on src/main/java/com/hgutierrezg/training/model/Timesheet.java:
```diff
@@ -0,0 +1,51 @@
+            return false;
+        Timesheet other = (Timesheet) obj;
+        return id.equals(other.id);
+    }
```
avoid overriding equals unless you have a functional need to do so.

Mbd06b commented on src/main/java/com/hgutierrezg/training/service/TimesheetService.java:
```diff
@@ -0,0 +1,38 @@
+@Service
+@AllArgsConstructor
+public class TimesheetService {
+    private final TimesheetRepository timesheetRepository;
```
You have a TimesheetRepository bean annotated as an @Repository, you should be able to Autowire this in the Service.

hgutierrezg replied:
Here spring is doing the injection too, not with the Autowire annotation but with the constructor injection. 
The constructor is generated by lombok with AllArgsConstructor.

chriswbluer commented on src/main/java/com/hgutierrezg/training/repository/TimesheetRepository.java:
```diff
@@ -0,0 +1,87 @@
+        this.timesheetList = MockedTimesheetsBuilder();
+    }
+
+    /**
```
nice comments

Mbd06b commented on src/main/java/filters/CorsFilter.java:
```diff
@@ -0,0 +1,31 @@
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletResponse;
+
+ public class CorsFilter implements Filter {
```
this shouldn't be necessary for locally hosted js communicating with the same server.

Mbd06b commented on src/main/webapp/components/dashboard/dashboard.html:
```diff
@@ -0,0 +1,30 @@
+        <div class="col-sm-3 sidenav">
+            <h4>Shared Timesheet</h4>
+            <ul class="nav nav-pills nav-stacked">
+                <li ng-click="dashboardController.menuSelection('employee') "
```
508 Accessibility Compliance.  All interactive components need a unique ID. 
https://www.deque.com/blog/unique-id-attributes-matter/

Mbd06b commented on src/main/webapp/components/dashboard/dashboard.html:
```diff
@@ -0,0 +1,30 @@
+<div class="container-fluid">
+    <div class="row content">
+        <div class="col-sm-3 sidenav">
```
508 Accessibility Compliance. use a `<nav>` semantic html5 element to describe your navigation. https://www.w3schools.com/html/html5_semantic_elements.asp

Mbd06b commented on src/main/webapp/components/dashboard/dashboard.html:
```diff
@@ -0,0 +1,30 @@
+                <li ng-click="dashboardController.menuSelection('employee') "
+                    ng-class="{ 'active' : dashboardController.role == 'employee' }"><a href="#employee">Employee</a></li>
+
+                <li ng-click="dashboardController.menuSelection('internalManager') "
+                    ng-class="{ 'active' : dashboardController.role == 'internalManager' }"><a href="#internalManager">Internal
```
508  Accessibility compliance.  ng-click and an 'a' tag. This is a case of "nested interactive elements" which will cause issues. 
https://accessibilityinsights.io/info-examples/web/nested-interactive/

Mbd06b commented on src/main/webapp/components/timesheet-form/timesheet-form.component.js:
```diff
@@ -0,0 +1,54 @@
+/**
+ * Creating controller only inside sharedTimesheetApp module,
```
Just note at a high level, while components are good to understand, particularly for UI pattern development. I would be sure to study the Module -> Controller -> Service  components of AngularJS to get a clear familiarity with the application architecture you might encounter.

Mbd06b commented on src/main/webapp/components/timesheet-list/timesheet-list.html:
```diff
@@ -0,0 +1,38 @@
+                <span ng-class="time.approved == true ? 'bi bi-check-lg': 'bi bi-x-lg' "></span>
+            </td>
+            <td class="approveBtn">
+                <button type="button"
+                        ng-if="time.approved == true && timesheetListController.role == 'internalManager'"
+                        ng-click="timesheetListController.approve(time)"
+                        class="btn btn-success custom-width">Approve
```
id attribute

Mbd06b commented on src/main/webapp/service/timesheet.service.js:
```diff
@@ -0,0 +1,57 @@
+    .module('sharedTimesheetApp')
+    .service('timesheetService', timesheetService);
+
+timesheetService.$inject = ['$http', '$q'];
+
+function timesheetService($http, $q) {
```
consider refactoring this service using promise syntax, you can completely avoid using $q defer library dependency.

Mbd06b reviewed (changes requested):
The only requirement I see missing is demonstrating the Synchronous form submit, with a required and optional field.  I reccomend adding a simple userRegistration.jsp that you can bind a form to and display a a property with that form submit with `${  }`  reference in a header element of your timesheet dashboard.

heidygutierrez committed:
Fix to nested link for accessibility.
Refactor to angular service
508 accessibility improvements by adding section and nav tags
Adding html unique ids

heidygutierrez committed:
Addressing pr on task 1 comments
Adding Login form with static call to login end point

heidygutierrez committed:
Deleting needed method

heidygutierrez committed:
Renaming labels

heidygutierrez committed:
Renaming variable in timesheet list

heidygutierrez committed:
Refactor to load angular context from jsp page.
Removing components

heidygutierrez committed:
Moving angular resources to static folder

heidygutierrez committed:
Moving angular resources to static folder
